src/java/org/apache/cassandra/service/paxos/v1/PrepareCallback.java

```diff
@@ -79,8 +86,28 @@ public synchronized void onResponse(Message<PrepareResponse> message)
         latch.decrement();
     }
 
-    public Iterable<InetAddressAndPort> replicasMissingMostRecentCommit()
+    public Iterable<InetAddressAndPort> replicasMissingMostRecentCommit(TableMetadata metadata)
     {
+        /**
+         * this check is only needed for mixed mode operation with 4.0 and can be removed once upgrade support dropped
+         * see the comment in {@link org.apache.cassandra.distributed.upgrade.MixedModePaxosTTLTest} for a full explanation.
+         */
+        if (DatabaseDescriptor.paxosStatePurging() == Config.PaxosStatePurging.legacy)
```

Comment:
I wonder how you safely transition to a new Paxos state purging configuration? It's not atomic to change it across the entire cluster.